daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -816,11 +817,14 @@ private[lf] object SBuiltin {
   }
 
   /** $tcon[fields] :: a -> b -> ... -> Struct */
-  final case class SBStructCon(fields: ImmArray[Name])
-      extends SBuiltinPure(fields.length)
+  final case class SBStructCon(inputFieldsOrder: Struct[Int])
```

Comment:
Maybe worth adding a comment that the reason we need to preserve the order is to preserve evaluation order?

Reply:
done